src/DynamoCore/Models/PresetsState.cs

```diff
@@ -0,0 +1,109 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Dynamo.Core;
+using Dynamo.Services;
+using System.Xml;
+using Dynamo.Nodes;
+
+namespace Dynamo.Models
+{
+    /// <summary>
+    /// a class that saves the state of a subset of a graph
+    /// </summary>
+    public class PresetState
```

Comment:
This is a `public` class.  Let's make sure your documentation is carefully written.  Remember that it's most important for your code to be readable.  This means it can be more easily maintained.